src/EoS/Nuclear/findtemp.cpp

```diff
@@ -0,0 +1,337 @@
+#include "NuclearEoS.h"
+
+#if ( MODEL == HYDRO  &&  EOS == EOS_NUCLEAR )
+
+
+
+GPU_DEVICE static
+void findtemp_bdry( const real x, const real y, const real z, 
+                     real *found_lt, const real *alltables_mode,
+                     const int nx, const int ny, const int nz, const int ntemp, 
+                     const real *xt, const real *yt, const real *zt, 
+                     const real *logtemp, const int keymode, int *keyerr );
+#ifdef __CUDACC__
+GPU_DEVICE static
+void findtemp( const real x, const real y, const real z, 
+                real *found_lt, const real *alltables_mode,
+                const int nx, const int ny, const int nz, const int ntemp, 
+                const real *xt, const real *yt, const real *zt, 
+                const real *logtemp, const int keymode, int *keyerr );
+#endif
+
+
+
+
+//-------------------------------------------------------------------------------------
+// Function    :  findtemp
+// Description :  Finding energy from different modes
+//                -->                 energy mode   (0)
+//                                    entropy mode  (2)
+//                                    pressure mode (3)
+//                Using 3D Catmull-Rom cubic interpolation formula
+//                to search the corresponding temperature given (rho, (eps, e, P), Y_e)
+//
+// Note        :  1. Use 3D Catmull-Rom cubic interpolation formula
+//                   to search the corresponding energy given (rho, (eps, e, P), Y_e)
+//                2. Invoked by nuc_eos_C_short()
+//
+// Parameter   :  x              : input vector of first  variable (rho)
+//                y              : input vector of second variable (eps, e, P)
+//                z              : input vector of third  variable (Y_e)
+//                found_lt       : output log(temp) of interpolated function values
+//                alltables_mode : 3d array of tabulated logtemp
+//                nx             : x-dimension of table
+//                ny             : y-dimension of table
+//                nz             : z-dimension of table
+//                ntemp          : size of temperature array in the Nuclear Eos table
+//                xt             : vector of x-coordinates of table
+//                yt             : vector of y-coordinates of table
+//                zt             : vector of z-coordinates of table
+//                logtemp        : log(T) array in the table
+//                keymode        : Which mode we will use
+//                                 --> 1: Energy mode   (coming in with internal energy)
+//                                     2: Entropy mode  (coming in with entropy)
+//                                     3: Pressure mode (coming in with P)
+//                keyerr         : output error
```

Comment:
- Format is not consistent with `findenergy.cpp`
  - e.g., upper/lower case of leading characters
- missing description for return parameters

- same questions for `findtemp_brdy()`

Reply:
I corrected all. please check.

Reply:
I checked above comments.